api/http-api.md

```diff
@@ -0,0 +1,173 @@
+  * `RespondActivityTaskCompletedById` - `POST /api/v1/namespaces/{namespace}/activities/complete-by-id`
+  * `RespondActivityTaskFailed` - `POST /api/v1/namespaces/{namespace}/activities/fail`
+  * `RespondActivityTaskFailedById` - `POST /api/v1/namespaces/{namespace}/activities/fail-by-id`
+  * `RespondActivityTaskCanceled` - `POST /api/v1/namespaces/{namespace}/activities/cancel`
+  * `RespondActivityTaskCanceledById` - `POST /api/v1/namespaces/{namespace}/activities/cancel-by-id`
+* Search attributes:
+  * Intentionally not exposed high-level search attribute calls from workflow service
+* Task polling/completing:
+  * Intentionally not exposed (except for activity task stuff needed for async activities)
+* Task queues:
+  * `ResetStickyTaskQueue` - Intentionally not exposed
+  * `DescribeTaskQueue` - `GET /api/v1/namespaces/{namespace}/task-queues/{task_queue_name}`
+  * `ListTaskQueuePartitions` - Intentionally not exposed
+* Schedules:
+  * `CreateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `DescribeSchedule` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `UpdateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/update`
+    * This is not a `PUT` because update different from create and not `PATCH` to avoid confusion with below
+  * `PatchSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/patch`
+    * This is not a `PATCH` to avoid confusion with update above
+  * `ListScheduleMatchingTimes` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}/matching-times`
+  * `DeleteSchedule` - `DELETE /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+    * Confirm whether eventually consistent
+  * `ListSchedules` - `GET /api/v1/namespaces/{namespace}/schedules`
+* Batch operations:
+  * `StartBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `DescribeBatchOperation` - `GET /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `ListBatchOperations` - `GET /api/v1/namespaces/{namespace}/batch-operations`
+  * `StopBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}/stop`
+* Worker build IDs:
+  * Intentionally not exposed until they settle
+* Other calls:
+  * `GetClusterInfo` - Intentionally not exposed
+  * `GetSystemInfo` - Intentionally not exposed
+
+### Embedding into Server
+
+Current suggested approach, subject to discussion:
+
+* Generated gRPC gateway code will be auto-generated at the same time as api-go is for gRPC and live alongside it
+* Service config has new `httpPort`
+  * Only allowed on "frontend" service, presence elsewhere is config failure
+  * Default value is `grpcPort` + 10
+    * Makes `7243` the default - unassigned in IANA, unused in server
+    * Cannot think of case where +10 clashes, e.g. if they use 80 or 443 or 8080 or whatever, 90, 453, and 8090 are fine
```

Comment:
I don't think these are commonly used ports for Temporal. I bet nobody customizes the grpc port apart from our devs that want to run multiple servers for testing purposes.

Reply:
So even a stronger statement that default of `grpcPort + 10` is ok :-)

Reply:
Now that I'm thinking of it, maybe 8233 is a better port since we already use that for the single binary UI server.

Reply:
That's what makes it a bad port. We use it for something else. And this is gonna be embedded in the server, so the +10 still keeps the port in the small range of ports we open there.

Reply:
We need to make a call if this API is going to serve the UI or not.
Seems like you're against it for some reason.

Reply:
> We need to make a call if this API is going to serve the UI or not.

I don't think we need to make this call. If needed I can add as a non-goal "decide how UI is going to access server" to make it more clear.

> Seems like you're against it for some reason

I'm not, I just don't think it needs to infect any work here. But if they want to be a consumer of this like anyone else, no problem. No UI specific stuff needed in this proposal though.

Reply:
I think we should let someone on the UI team review this proposal and comment here.

> Nothing in this proposal precludes potential UI use or potential TypeScript use, they are just distractions from an initial version.

TS is for sure going to be a heavy consumer of this API and maybe UI too.
I don't think these are distractions, they're requirements.
We don't want to add another HTTP API if we realize that this one can't be used for purposes that we've foreseen.

Reply:
What in this proposal is stopping TS or UI use? You can answer for TS and will wait for UI review.

Reply:
For TS I'd want to know how the translation of HTTP bodies <-> gRPC messages would work.
If it's standard proto3 JSON (with the acceptable exception of our custom enum encoding) you have a go from the TS SDK perspective.

Looks like the encoding is [configurable](https://github.com/grpc-ecosystem/grpc-gateway#mapping-grpc-to-http) so we should be good, but let's please clarify in the proposal.

Reply:
There are multiple places where our API is not the cleanest and/or won't translate as well to JSON as it would if we _designed_ a REST API. But I think that is out of scope for this simple project which is just make HTTP and gRPC both available for the same API.

Reply:
Resolving as no actionable suggestions and/or beyond the goals